Tools/RemoteConsole/ly_remote_console/ly_remote_console/remote_console_commands.py

```diff
@@ -221,78 +220,87 @@ def pump(self):
-            'Returning "{}" for expect_log_line() - previously this returned a function object, '
-            'so if you see failures now this may be why.'.format(event_success))
+        logger.warning(f'Returning "{event_success}" for expect_log_line() - previously this returned a function object, so if you see failures now this may be why.')
         return event_success
 
-    def _create_message(self, message_type, message_body=''):
-        # type: (bytes, str) -> bytearray
+    def _create_message(self, message_type: int, message_body: str='') -> bytearray:
         """
         Transforms a message to be sent to the launcher. The string is converted to a bytearray and
         is prepended with the message type and appended with an ending 0.
         :param message_type: Uses CONSOLE_MESSAGE_MAP to prepend the bytearray message
         :param message_body: The message string to be converted
         """
-        message_body = message_body.encode()
         message = bytearray(0)
 
         message.append(message_type)
 
+        message_body = message_body.encode()
         for message_body_char in message_body:
             message.append(message_body_char)
 
         message.append(0)
         return message
 
-    def _send_message(self, message):
-        # type: (bytearray) -> None
+    def _send_message(self, message: bytearray):
         """
         Sends console commands through the socket connection to the launcher. The message string should
         first be transformed into a bytearray.
         :param message: The message to be sent to the Launcher instance
         """
         self.socket.sendall(message)
 
-    def _handle_message(self, message):
-        # type: (bytearray) -> None
+    def _handle_message(self, message: bytearray):
         """
         Handles the messages and and will poll for expected console messages that we are looking for and set() events to True.
         Displays the message if we determine it is a logging message.
         :param message: The message (a byte array) received to be handled in various ways
         """
+        if len(message) < 1:
+            logger.error('received an empty message')
+            print(f'Error: received an empty message')
+            return
+
         # message[0] is the representation of the message type inside of the message received from our launchers
         message_type = message[0]
 
         # ignoring the first byte and the last byte.  The first byte is the message type and the last byte is a
         # Null terminator
         message_body = message[1:-1]
 
+        '''
+        diagnostic code
+        if message_type != 49 and message_type != 50:
+            if message_type in REVERSE_CONSOLE_MESSAGE_MAP:
+                print(f'handle message {message_type}/{REVERSE_CONSOLE_MESSAGE_MAP[message_type]} : {message[1:]}')
+            else:
+                print(f'handle message {message_type}/unknown : {message[1:]}')
+        '''
+        
```

Comment:
What is this diagnosing?  What does 49 & 50 mean?   If it's just for your purposes, rather than to catch & print something, please remove.

Reply:
49 is eCET_Req, it is a service message
50 is eCET_LogMessage, it is a log string
This code piece is to debug issues, like log suddenly stopped, but what happened?
The numbers come from enum EConsoleEventType
In Pythion it is '0' + enum id, so 48, 49, 50 etc. The type is the first character of the message.
I want to keep this code because I predict issues like I describer above, so I'll have to investigate them.

Reply:
Gotcha - then please comment "why" and also define the numbers so it's readable and understandable.